components/profile/ProfileModal.tsx

```diff
@@ -1,105 +1,866 @@
+        if (!pendingBid || BigNumber.from(pendingBid.contributionRate).eq(0)) {
+          status = "Valid";
+          actionDate = "";
+          forSalePrice = BigNumber.from(currentBid.forSalePrice);
+          perSecondFeeNumerator = BigNumber.from(
+            currentBid.perSecondFeeNumerator
+          );
+          action = PortfolioAction.VIEW;
+        } else if (BigNumber.from(pendingBid.contributionRate).gt(0)) {
+          status = "Incoming Bid";
+          actionDate = dayjs
+            .unix(pendingBid.timestamp)
+            .add(7, "day")
+            .format("YYYY/MM/D");
+          forSalePrice = BigNumber.from(pendingBid.forSalePrice);
+          perSecondFeeNumerator = BigNumber.from(
+            pendingBid.perSecondFeeNumerator
+          );
+          action = PortfolioAction.RESPOND;
+        } else {
+          continue;
+        }
+      } else if (
+        pendingBid &&
+        BigNumber.from(pendingBid.contributionRate).gt(0)
+      ) {
+        status = "Outgoing Bid";
+        actionDate = dayjs
+          .unix(pendingBid.timestamp)
+          .add(7, "day")
+          .format("YYYY/MM/D");
+        forSalePrice = BigNumber.from(pendingBid.forSalePrice);
+        perSecondFeeNumerator = BigNumber.from(
+          pendingBid.perSecondFeeNumerator
+        );
+        action = PortfolioAction.VIEW;
+      } else {
+        continue;
+      }
+
+      annualFee = forSalePrice.div(perSecondFeeNumerator);
+      buffer = calculateBufferNeeded(annualFee, NETWORK_ID);
+
+      const assetContentManager = getAssetContentManager(
+        parcelId,
+        licenseOwner
+      ) as AssetContentManager;
+      const basicProfileStreamManager = new BasicProfileStreamManager(
+        assetContentManager
+      );
+
+      const promiseChain = licenseDiamondContract
+        .contributionRate()
+        .then((ownerBidContributionRate) => {
+          if (pendingBid && BigNumber.from(pendingBid.contributionRate).gt(0)) {
+            const deadline = Number(pendingBid.timestamp) + SECONDS_IN_WEEK;
+            const isPastDeadline = deadline * 1000 <= Date.now();
+            if (isPastDeadline || ownerBidContributionRate.eq(0)) {
+              status = "Needs Tranfer";
+              actionDate = dayjs
+                .unix(
+                  deadline < Number(currentBid.timestamp)
+                    ? deadline
+                    : currentBid.timestamp
+                )
+                .format("YYYY/MM/D");
+              forSalePrice = BigNumber.from(pendingBid.forSalePrice);
+              action = PortfolioAction.TRIGGER;
+            }
+          }
+        })
+        .then(() => licenseDiamondContract.isPayerBidActive())
+        .then((isPayerBidActive) => {
+          if (licenseOwner === account && !isPayerBidActive) {
+            status = "In Foreclosure";
+            actionDate = dayjs.unix(currentBid.timestamp).format("YYYY/MM/D");
+            forSalePrice = calculateAuctionValue(
```

Comment:
The logic for calculating the current foreclosure auction price has been updated https://github.com/Geo-Web-Project/cadastre/blob/develop/components/cards/AuctionInfo.tsx#L80

This may be updated again soon, so we should probably put it in a shared place.

Reply:
I put `calculateAuctionValue()` in `/lib/utils.ts`, is that ok?

Reply:
That is fine, but see the change in logic. The auction start time and length are different now.

Reply:
I see, should I write a function that returns an object with start time and auction length to be reused or just use the same logic here?

Reply:
You could keep `calculateAuctionValue` as is I suppose, but just update this to pass in the correct start time and length. But the logic for getting the start time and length should be defined in a single place.